library/default-synths-extra.scd

```diff
@@ -523,3 +523,148 @@ SynthDef(\superreese,  {|out, pan, freq, sustain, accelerate, detune=0, voice=0|
 	OffsetOut.ar(out, DirtPan.ar(sound, ~dirt.numChannels, pan, env));
 }).add;
 );
+
+(
+// DrumSynths SC Example - SOS Drums by Renick Bell, renick@gmail.com
+// recipes from Gordon Reid in his Sound on Sound articles
+// SOSkick — http:www.soundonsound.com/sos/jan02/articles/synthsecrets0102.asp
+// increase pitch1 and voice for interesting electronic percussion
+// Adapted to SuperDirt by Aleksandr Yakunichev, hi@ya.codes
+//
+// midinote – controls the root note of the kick,
+//            but wraped between c2 (24) and c3 (36) notes
+//            to avoid unpleasant sounds
+// pitch1 - controls modulation frequency in Hz (min: 0, max: SampleRate.ir / 2)
+// voice - controls modulation input phase in radians (min: 0, max: your sanity)
+// pitch2 - controls WhiteNoise amplitude (min: 0, max: 1)
+// speed - controls WhiteNoise sweep (min: 0, max: 1)
+SynthDef(\soskick, {
+  arg out, pan, midinote = 24, pitch1 = 0, speed = 0.3, voice = 1, sustain = 1, pitch2 = 0.00;
+  var beater, source, freq, env;
+
+  freq = (Wrap.ir(midinote, 24, 36)).midicps;
```

Comment:
Using midinote as an argument won't allow you to use `# freq` in tidal. Perhaps it would be better if you used `freq` as an argument directly, and then limit that by
`freq = freq.clip(24.midicps, 36.midicps)`
For this, I used clip, but of course if you have reasons for using wrap, that would also be rasonable ;-)

Reply:
Sure, didn't know about `freq` parameter in Tidal, thanks!

I'm using `Wrap.ir` for nice usage in Tidal: `d1 $ s "soskick" # n "e"`. Where you can just plug any note without octave specification in `n` param and it will adopt kick to it.

Reply:
ah I see, very good. So the best would be to define a parameter `root = 24`, which is the reference midinote, and then write `freq = freq.wrap(root.midicps, (root + 12).midicps)`

Reply:
… then you could work in different harmonic reference frames.